core/src/main/java/org/mule/runtime/core/exception/TemplateOnErrorHandler.java

```diff
@@ -110,11 +110,11 @@ protected MuleEvent processCatch(MuleEvent event, MessagingException exception)
         TransactionCoordination.getInstance().rollbackCurrentTransaction();
       } catch (Exception ex) {
         //Do nothing
+        logger.warn(ex.getMessage());
       }
 
-      event.setError(builder(exception).build());
       event.setMessage(MuleMessage.builder(event.getMessage()).exceptionPayload(new DefaultExceptionPayload(exception)).build());
-      return event;
+      throw exception;
```

Comment:
This is a very corner case that almost never happen. What was happening today was that the error handler was actually not failing when there's was an MP failing inside it. Since we are introducing the concept or an scope for error handling, if the users uses an MP that may fail, then they can use the scope to catch such exception. Worst case if there's no scope for error handling then the exception will be propagated to the parent caller.